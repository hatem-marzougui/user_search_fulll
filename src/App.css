.App {
  text-align: center;
  padding: 10px;
  display: flex;
  flex-direction: column;
  gap: 5px;
}
