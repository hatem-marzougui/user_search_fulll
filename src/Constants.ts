export const HEADER_TITLE: string = "Github Search";
export const VIEW_PROFILE_BUTTON_TEXT: string = "View Profile";
export const LOGIN: string = "Login";
export const EDIT_MODE: string = "Edit Mode";
export const DUPLICATE_USERS: string = "Duplicate Users";
export const SEARCH_PLACEHOLDER: string = "Search...";
export const USERS_API_ENDPOINT: string = "/search/users";
export const RATE_LIMIT_ERROR_MESSAGE: string =
  "GitHub request limit reached. Please try again in a minute.";
export const EMPTY_SEARCH_QUERY_ERROR: string = "Please enter a search query";
export const NO_RESULTS_FOUND: string = "No result found  ";
export const USER_CARD_AVATAR_ALT: string = "User Avatar";
export const USER_CARD_ID_LABEL: string = "ID:";
export const USER_CARD_LOGIN_LABEL: string = "Login:";
export const USER_CARD_VIEW_PROFILE_BUTTON_TEXT: string = "View Profile";
export const RATE_LIMIT: string = "rate limit";
export const ELEMENT_SELECTED: string = "elements selected";
