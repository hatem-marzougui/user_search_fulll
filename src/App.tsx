import React from "react";
import "./App.css";
import Header from "./Components/Header/Header";
import UsersList from "./Components/UsersList/UsersList";

const App: React.FC = () => {
  return (
    <div className="App">
      <Header />
      <UsersList />
    </div>
  );
};

export default App;
