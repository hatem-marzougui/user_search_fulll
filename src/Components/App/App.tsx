import React from "react";
import "./App.css";
import Header from "../Header/Header";
import UsersList from "../UsersList/UsersList";

const App: React.FC = () => {
  return (
    <div className="App">
      <Header />
      <UsersList />
    </div>
  );
};

export default App;
