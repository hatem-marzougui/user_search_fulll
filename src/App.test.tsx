import { render, screen } from "@testing-library/react";
import App from "./App";
import { HEADER_TITLE, SEARCH_PLACEHOLDER } from "./Constants";

test("App must renders header", () => {
  render(<App />);
  const headerElement = screen.getByText(HEADER_TITLE);
  expect(headerElement).toBeInTheDocument();
});

test("App must renders search input", () => {
  render(<App />);
  const searchInput = screen.getByPlaceholderText(SEARCH_PLACEHOLDER);
  expect(searchInput).toBeInTheDocument();
});

test("App must renders users list", () => {
  render(<App />);
  const usersList = screen.getByTestId("users-list");
  expect(usersList).toBeInTheDocument();
});
test("matches snapshot", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
