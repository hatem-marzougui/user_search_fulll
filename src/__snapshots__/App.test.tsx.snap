// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="headerContainer"
    >
      <b>
        Github Search
      </b>
    </div>
    <div
      class="search-container"
    >
      <input
        class="search-input"
        placeholder="Search..."
        type="search"
        value=""
      />
    </div>
    <div
      class="control-div"
    >
      <div
        class="users-count"
      >
        <img
          alt="liste"
          class="icon"
          height="24"
          src="/icons/list.svg"
          width="24"
        />
        <span>
          0
           
          elements selected
        </span>
      </div>
      <div
        class="controls-buttons"
      />
    </div>
    <div
      class="users-list-container"
      data-testid="users-list"
    >
      <div
        class="message"
      >
        Please enter a search query
      </div>
    </div>
  </div>
</div>
`;
